code/modules/research/xenobiology/crossbreeding/_structures.dm

```diff
@@ -0,0 +1,667 @@
+	for(var/turf/T in RANGE_TURFS(4,src))
+		T.add_atom_colour(get_colour(), FIXED_COLOUR_PRIORITY)
+
+	addtimer(CALLBACK(src,.proc/change_colour),rand(0.75 SECONDS,1.25 SECONDS))
+
+/obj/structure/slime_crystal/pyrite/proc/get_colour()
+	var/itemcolor = "#FFFFFF"
+	var/C = rand(0,19)
+	switch(C)
+		if(0)
+			itemcolor = "#FFA500"
+		if(1)
+			itemcolor = "#B19CD9"
+		if(2)
+			itemcolor = "#ADD8E6"
+		if(3)
+			itemcolor = "#7E7E7E"
+		if(4)
+			itemcolor = "#FFFF00"
+		if(5)
+			itemcolor = "#551A8B"
+		if(6)
+			itemcolor = "#0000FF"
+		if(7)
+			itemcolor = "#D3D3D3"
+		if(8)
+			itemcolor = "#32CD32"
+		if(9)
+			itemcolor = "#704214"
+		if(10)
+			itemcolor = "#2956B2"
+		if(11)
+			itemcolor = "#FAFAD2"
+		if(12)
+			itemcolor = "#FF0000"
+		if(13)
+			itemcolor = "#00FF00"
+		if(14)
+			itemcolor = "#FF69B4"
+		if(15)
+			itemcolor = "#FFD700"
+		if(16)
+			itemcolor = "#505050"
+		if(17)
+			itemcolor = "#000000"
+		if(18)
+			itemcolor = "#FFB6C1"
+		if(19)
+			itemcolor = "#008B8B"
+	return itemcolor
+
+/obj/structure/slime_crystal/red
+	colour = "red"
+
+	var/blood_amt = 0
+
+	var/max_blood_amt = 300
+
+/obj/structure/slime_crystal/red/examine(mob/user)
+	. = ..()
+	. += "It has [blood_amt] u of blood."
+
+/obj/structure/slime_crystal/red/process()
+
+	if(blood_amt == max_blood_amt)
+		return
+
+	for(var/obj/effect/decal/cleanable/blood/B in range(3,src))
+		qdel(B)
+
+		blood_amt++
+		if(blood_amt == max_blood_amt)
+			return
+
+/obj/structure/slime_crystal/red/attack_hand(mob/user)
+	if(blood_amt < 100)
+		return ..()
+
+	blood_amt -= 100
+	var/type = pick(/obj/item/reagent_containers/food/snacks/meat/slab,/obj/item/organ/heart,/obj/item/organ/lungs,/obj/item/organ/liver,/obj/item/organ/eyes,/obj/item/organ/tongue,/obj/item/organ/stomach,/obj/item/organ/ears)
+	new type(get_turf(src))
+
+/obj/structure/slime_crystal/red/attacked_by(obj/item/I, mob/living/user)
+	if(blood_amt < 10)
+		return ..()
+
+	if(!istype(I,/obj/item/reagent_containers/glass/beaker))
+		return ..()
+
+	var/obj/item/reagent_containers/glass/beaker/item_beaker = I
+
+	if(!item_beaker.is_refillable() || (item_beaker.reagents.total_volume + 10 > item_beaker.reagents.maximum_volume))
+		return ..()
+	blood_amt -= 10
+	item_beaker.reagents.add_reagent(/datum/reagent/blood,10)
+
+/obj/structure/slime_crystal/green
+	colour = "green"
+	var/datum/mutation/stored_mutation
+
+/obj/structure/slime_crystal/green/examine(mob/user)
+	. = ..()
+	if(stored_mutation)
+		. += "It currently stores [stored_mutation.name]"
+	else
+		. += "It doesn't hold any mutations"
+
+/obj/structure/slime_crystal/green/attack_hand(mob/user)
+	. = ..()
+	if(!ishuman(user))
+		return
+	var/mob/living/carbon/human/human_user = user
+	var/list/mutation_list = human_user.dna.mutations
+	stored_mutation = pick(mutation_list)
+	stored_mutation = stored_mutation.type
+
+/obj/structure/slime_crystal/green/on_mob_effect(mob/living/affected_mob)
+	. = ..()
+	if(!ishuman(affected_mob) || !stored_mutation)
```

Comment:
You should check for their DNA rather than if they are human, I believe monkeys can have DNA to. Also how does this interact with people with fucked DNA due to changelings husking people, or does it not work for dead bodies?